cmd/util/testutils/cmd.go

```diff
@@ -0,0 +1,94 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package testutils
+
+import (
+	"bytes"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	durationExpr = `duration.*\)`
+	durationRepl = "duration )"
+)
+
+// Args represent are required args to test a command.
+type Args struct {
+	Cmd  *cobra.Command
+	Args []string
+	Cfg  MockCfg
+}
+
+// Assertion to use for tests.
+type Assertion struct {
+	Err    error
+	Stdout string
+	Stderr string
+}
+
+// TestCommand tests a `*cobra.Command` and uses the testing.T struct to
+// return any unmatched assertions.
+func TestCommand(t *testing.T, args Args, assertion Assertion) {
```

Comment:
I'm not sure I like `testutils.TestCommand` 🤔 How about `cmdutils.Test` or `testutils.AssertCommand` ? or something better that you might come up with 😄

Reply:
haha fair point, maybe `testutils.RunAssertion`, or just `Run`? `cmdutils.TestCommand` sounds good too.

Reply:
I think just `Run` might be a bit confusing if the name of the package is `testutils`. I think I like `testutils.RunCmdAssertion` or `cmdutils.RunAssertion`. What do you think?